### ISSUE TITLE
RuntimeError('FirmwareError: firmware error') when trying to sign a simple transaction 

### ISSUE THREAD

**Issue by hakonamatata** [firoorg/electrum-firo#3]
Posted this in the Trezor Issue tracker, and they found an error in the code here:

[https://github.com/firoorg/electrum-firo/blob/21052243d57b3c51fbd6aecaf3353e1babef7454/electrum_dash/plugins/trezor/trezor.py#L210](https://github.com/firoorg/electrum-firo/blob/21052243d57b3c51fbd6aecaf3353e1babef7454/electrum_dash/plugins/trezor/trezor.py#L210)

Original post: [https://github.com/trezor/trezor-firmware/issues/1588](https://github.com/trezor/trezor-firmware/issues/1588)

I am starting to get a bit worried. I can not sign transactions and spend/send. I have tried one input to one output. One input to an output and a change address. Basic transactions. Nothing works, I get this firmware error every time.

**Firmware version and revision**

Using Trezor model T

Firmware version
2.3.6

**Desktop/smartphone setup (please complete the following information):**
Tried using Trezor Suite and Electrum, same issue. Does not matter what wallet I use. 

**To Reproduce**
Any transaction with Firo it seems. Fails with Firmware error every time. 

**Screenshots**
![image](https://user-images.githubusercontent.com/4003969/116810016-92db4680-ab41-11eb-8100-241fe652a223.png)

**Additional context**
I am using Firo [https://www.firo.org/](https://www.firo.org/)

All my coins are locked with Trezor, but if I can't spend that is a big problem.

**Comment by a-bezrukov** [comment]
Are you using a custom firmware?
Zcoin was rebranded to Firo on March 25 https://github.com/trezor/trezor-firmware/pull/1544#event-4507564057
The latest firmwares for Trezor are dated 15 February 2021 (Model T) and 10 February 2021 (Trezor One). So, this change should not be in the firmwares yet.

**Comment by hakonamatata** [comment]
> Are you using a custom firmware?
> Zcoin was rebranded to Firo on March 25 [trezor/trezor-firmware#1544 (comment)](https://github.com/trezor/trezor-firmware/pull/1544#event-4507564057)
> The latest firmwares for Trezor are dated 15 February 2021 (Model T) and 10 February 2021 (Trezor One). So, this change should not be in the firmwares yet.

Using the standard hardware. Do you know when these new changed will be added to a new firmware release?

**Comment by a-bezrukov** [comment]
I think with the new firmware release. I am waiting for a new firmware release to change it here (quite the same as your PR does)... My Trezor One works fine with the latest firmware and coin=Zcoin. I am moving on with investigating the issue. In the meanwhile, you can spend your coins, recreating a software wallet from the seed phrase as described here: https://github.com/firoorg/electrum-firo-archived/issues/21#issuecomment-472604529

**Comment by a-bezrukov** [comment]
@hakonamatata,
Could you run your electrum-firo with Trezor again, but this time please specify -vdebug flag for electrum-firo to produce a verbose debug output. Please post the relevant portion of logs right before the error happens.

**Comment by hakonamatata** [comment]
> 
> 
> @hakonamatata,
> Could you run your electrum-firo with Trezor again, but this time please specify -vdebug flag for electrum-firo to produce a verbose debug output. Please post the relevant portion of logs right before the error happens.

```
20210504T100514.385790Z |     INFO | network | fee_histogram []
20210504T100514.914843Z |     INFO | network | fee_estimates {25: 11456, 10: 11456, 5: 11456, 2: 11456}
20210504T100516.129137Z |     INFO | network | fee_estimates {25: 11456, 10: 11456, 5: 11456, 2: 11456}
20210504T100516.211297Z |     INFO | network | fee_estimates {25: 11456, 10: 11456, 5: 11456, 2: 11456}
20210504T100517.148671Z |     INFO | network | fee_estimates {25: 11456, 10: 11456, 5: 11456, 2: 11456}
20210504T100528.627519Z |     INFO | plugin.DeviceMgr | getting client for keystore
20210504T100528.627519Z |     INFO | plugin.DeviceMgr | scanning devices...
20210504T100529.226918Z |    DEBUG | util.profiler | DeviceMgr.scan_devices 0.5994
20210504T100529.226918Z |     INFO | plugin.DeviceMgr | end client for keystore
20210504T100529.391844Z |     INFO | plugin.DeviceMgr | getting client for keystore
20210504T100529.392837Z |     INFO | plugin.DeviceMgr | scanning devices...
20210504T100530.006084Z |    DEBUG | util.profiler | DeviceMgr.scan_devices 0.6132
20210504T100530.007076Z |     INFO | plugin.DeviceMgr | end client for keystore
20210504T100543.984048Z |    ERROR | gui.qt.main_window.[default_wallet] | on_error
Traceback (most recent call last):
  File "electrum_firo\plugins\trezor\clientbase.py", line 257, in sign_tx
  File "trezorlib\tools.py", line 231, in wrapped_f
  File "trezorlib\btc.py", line 248, in sign_tx
  File "trezorlib\tools.py", line 231, in wrapped_f
  File "trezorlib\client.py", line 223, in call
trezorlib.exceptions.TrezorFailure: FirmwareError: Firmware error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "electrum_firo\gui\qt\util.py", line 848, in run
  File "electrum_firo\wallet.py", line 1446, in sign_transaction
  File "electrum_firo\plugins\trezor\trezor.py", line 96, in sign_transaction
  File "electrum_firo\plugin.py", line 362, in wrapper
  File "electrum_firo\plugin.py", line 355, in run_in_hwd_thread
  File "concurrent\futures\_base.py", line 435, in result
  File "concurrent\futures\_base.py", line 384, in __get_result
  File "concurrent\futures\thread.py", line 57, in run
  File "electrum_firo\plugins\trezor\trezor.py", line 342, in sign_transaction
  File "electrum_firo\plugin.py", line 362, in wrapper
  File "electrum_firo\plugin.py", line 352, in run_in_hwd_thread
  File "electrum_firo\plugins\trezor\clientbase.py", line 257, in sign_tx
  File "electrum_firo\plugins\trezor\clientbase.py", line 85, in __exit__
RuntimeError: FirmwareError: Firmware error
```

**Comment by matejcik** [comment]
from @hakonamatata:
> Got another error message trying to send now:
> 
> ```
> Failed to send transaction
> Error details: Failed to decode message: memory allocation failed, allocating 13854 bytes
> ```

why is Firo Electrum sending 13-kilobyte messages to sign a 226-byte tx?

**Comment by a-bezrukov** [comment]
@hakonamatata , are you spending a Lelantus output?

**Comment by hakonamatata** [comment]
> 
> 
> @hakonamatata , are you spending a Lelantus output?

If I take this input [https://explorer.zcoin.io/address/a9CG27PviynRVdnLCDiyzZD4GJdMovuHGN](https://explorer.zcoin.io/address/a9CG27PviynRVdnLCDiyzZD4GJdMovuHGN)

and try to spend, I get the same firmware error. If that is what you are asking. 

But, I have tried to spend different inputs, always get the same error.

**Comment by justanwar** [comment]
Fixed in #8 and trezor/trezor-firmware/pull/1773
